### PR TITLE
Tsukuyomi Keyboard

### PR DESCRIPTION
**Pull request by awkannan**
## Description

Adding the Tsukuyomi keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/cannonkeys/tsukuyomi/tsukuyomi.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#pragma once
```
Comment (drashna) [inline keyboards/cannonkeys/tsukuyomi/tsukuyomi.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/cannonkeys/tsukuyomi/tsukuyomi.c [hunk 2]
```diff
@@ -0,0 +1 @@
+#include "tsukuyomi.h"
```
Comment (drashna) [inline keyboards/cannonkeys/tsukuyomi/tsukuyomi.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/cannonkeys/tsukuyomi/rules.mk [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+CONSOLE_ENABLE = yes	# Console for debug
+COMMAND_ENABLE = yes    # Commands for debug and configuration
+SLEEP_LED_ENABLE = yes  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes	    # USB Nkey Rollover
+CUSTOM_MATRIX = no # Custom matrix file
+BACKLIGHT_ENABLE = yes
+RGBLIGHT_ENABLE = yes
+WS2812_DRIVER = spi
+
+DFU_SUFFIX_ARGS =
```
Comment (drashna) [inline keyboards/cannonkeys/tsukuyomi/rules.mk:21]: Didn't we want `FFFF` for both the VID and PID here?

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/cannonkeys/tsukuyomi/rules.mk [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+CONSOLE_ENABLE = yes	# Console for debug
+COMMAND_ENABLE = yes    # Commands for debug and configuration
+SLEEP_LED_ENABLE = yes  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes	    # USB Nkey Rollover
+CUSTOM_MATRIX = no # Custom matrix file
+BACKLIGHT_ENABLE = yes
+RGBLIGHT_ENABLE = yes
+WS2812_DRIVER = spi
+
+DFU_SUFFIX_ARGS = -p FFFF -v FFFF
```
Comment (zvecr) [inline keyboards/cannonkeys/tsukuyomi/rules.mk:21]:
```suggestion
# Wildcard to allow APM32 MCU 
DFU_SUFFIX_ARGS = -p FFFF -v FFFF
```

File: keyboards/cannonkeys/tsukuyomi/info.json [hunk 5]
```diff
@@ -0,0 +1,12 @@
+{
+    "keyboard_name": "Tsukuyomi",
+    "url": "https://cannonkeys.com",
+    "maintainer": "awkannan",
+    "width": 17.25,
+    "height": 5,
+    "layouts": {
+        "LAYOUT": {
```
Comment (zvecr) [inline keyboards/cannonkeys/tsukuyomi/info.json:8]:
```suggestion
        "LAYOUT_default": {
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/cannonkeys/tsukuyomi/config.h [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+#pragma once
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xCA04
+#define PRODUCT_ID      0x0002
+#define DEVICE_VER      0x0001
+#define MANUFACTURER CannonKeys
+#define PRODUCT Tsukuyomi
+#define DESCRIPTION Tsukuyomi Keyboard
```
Comment (fauxpark) [inline keyboards/cannonkeys/tsukuyomi/config.h:26]:
```suggestion
```

File: keyboards/cannonkeys/tsukuyomi/keymaps/via/rules.mk [hunk 7]
```diff
@@ -0,0 +1,4 @@
+# rules.mk overrides to enable VIA
+
+VIA_ENABLE = yes
+
```
Comment (fauxpark) [inline keyboards/cannonkeys/tsukuyomi/keymaps/via/rules.mk:4]:
```suggestion
VIA_ENABLE = yes
```

File: keyboards/cannonkeys/tsukuyomi/readme.md [hunk 8]
```diff
@@ -0,0 +1,12 @@
+# Tsukuyomi
+
+Tsukuyomi Keyboard by persocom
+
+Keyboard Maintainer: [Andrew Kannan](https://github.com/awkannan)  
+Hardware Supported: STM32F072CBT6 / APM32F072CBT6
```
Comment (fauxpark) [inline keyboards/cannonkeys/tsukuyomi/readme.md:6]:
```suggestion
* Keyboard Maintainer: [Andrew Kannan](https://github.com/awkannan)
* Hardware Supported: STM32F072CBT6 / APM32F072CBT6
```

File: keyboards/cannonkeys/tsukuyomi/rules.mk [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+MOUSEKEY_ENABLE = yes	# Mouse keys
+EXTRAKEY_ENABLE = yes	# Audio control and System control
+CONSOLE_ENABLE = yes	# Console for debug
+COMMAND_ENABLE = yes    # Commands for debug and configuration
+SLEEP_LED_ENABLE = yes  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes	    # USB Nkey Rollover
+CUSTOM_MATRIX = no # Custom matrix file
+BACKLIGHT_ENABLE = yes
+RGBLIGHT_ENABLE = yes
+WS2812_DRIVER = spi
```
Comment (fauxpark) [inline keyboards/cannonkeys/tsukuyomi/rules.mk:19]:
```suggestion
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = yes      # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
WS2812_DRIVER = spi
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by zvecr — APPROVED** [review]

**Comment by zvecr** [comment]
Thanks!

**Pull request merged**